diff --git a/src/core/contexts/win32windowcontext.cpp b/src/core/contexts/win32windowcontext.cpp
index 9f21636..b0ab54a 100644
--- a/src/core/contexts/win32windowcontext.cpp
+++ b/src/core/contexts/win32windowcontext.cpp
@@ -2100,6 +2100,10 @@ namespace QWK {
         const auto clientRect = wParam ? &(reinterpret_cast<LPNCCALCSIZE_PARAMS>(lParam))->rgrc[0]
                                        : reinterpret_cast<LPRECT>(lParam);
         [[maybe_unused]] const auto &flickerReducer = qScopeGuard([this]() {
+            static const bool disabled =
+                qEnvironmentVariableIntValue("QWK_DISABLE_FLICKER_WORKAROUND") != 0;
+            if (disabled)
+                return;
             // When we receive this message, it means the window size has changed
             // already, and it seems this message always come before any client
             // area size notifications (eg. WM_WINDOWPOSCHANGED and WM_SIZE). Let
